apps/src/lib/levelbuilder/lesson-editor/ProgrammingExpressionsEditor.jsx

```diff
@@ -177,6 +197,18 @@ class ProgrammingExpressionsEditor extends Component {
           <Table.Header />
           <Table.Body rows={this.props.programmingExpressions} rowKey="id" />
         </Table.Provider>
+        <Button
+          text={'Add Introduced Code'}
+          onClick={this.handleOpenAddProgrammingExpression}
+          color={Button.ButtonColor.orange}
+        />
+        <FindProgrammingExpressionDialog
```

Comment:
This could be something to fix in a future PR but when I try to add multiple programming expressions to a lesson it seems to save some of the state. Trying to get a good screen capture

Reply:
This is the best I can get. The search seems to be saved but the search term is cleared

https://user-images.githubusercontent.com/46464143/112677236-b4e8f700-8e26-11eb-8c7a-f7d7c3eabcaf.mov

Reply:
Oh I don't think I thought about dealing with you clicking multiple. Will look.

Reply:
this is a property of the dialog itself, not of any of the code in this PR; it just requires binding state a bit closer to the value of the input rather than just listening for change events. I figured it wasn't a big enough deal, though, since the actual selection of the blocks still works